src/instancemanager.jl

```diff
@@ -0,0 +1,330 @@
+
+@enum InstanceManagerState NoSolver EmptySolver AttachedSolver
+@enum InstanceManagerMode Manual Automatic
+
+# TODO: Rename AbstractStandaloneInstance to AbstractInstance and AbstractSolverInstance to AbstractSolver?
```

Comment:
The name `AbstractInstance` is already taken by the supertype

Reply:
That's not a particularly strong argument against renaming.

Reply:
No, but we need to find a name for the supertype :-P

Reply:
Yes, but let's leave that discussion for later.